lib/binance/futures.ex

```diff
@@ -0,0 +1,372 @@
+
+  @spec create_listen_key(map()) :: {:ok, map()} | {:error, error()}
+  def create_listen_key(
+        receiving_window \\ 1500,
+        timestamp \\ nil,
+        config \\ nil
+      ) do
+    timestamp =
+      case timestamp do
+        # timestamp needs to be in milliseconds
+        nil ->
+          :os.system_time(:millisecond)
+
+        t ->
+          t
+      end
+
+    arguments = %{
+      timestamp: timestamp,
+      recvWindow: receiving_window
+    }
+
+    case HTTPClient.post_binance("/fapi/v1/listenKey", arguments, config) do
+      {:ok, %{"code" => code, "msg" => msg}} ->
+        {:error, {:binance_error, %{code: code, msg: msg}}}
+
+      data ->
+        data
+    end
+  end
+
+  @spec keep_alive_listen_key(integer(), integer() | nil, map() | nil) ::
+          {:ok, %{}} | {:error, error()}
+  def keep_alive_listen_key(receiving_window \\ 1500, timestamp \\ nil, config \\ nil) do
+    timestamp =
+      case timestamp do
+        # timestamp needs to be in milliseconds
+        nil ->
+          :os.system_time(:millisecond)
+
+        t ->
+          t
+      end
+
+    arguments = %{
+      timestamp: timestamp,
+      recvWindow: receiving_window
+    }
+
+    case HTTPClient.put_binance("/fapi/v1/listenKey", arguments, config) do
+      {:ok, %{"code" => code, "msg" => msg}} ->
+        {:error, {:binance_error, %{code: code, msg: msg}}}
+
+      data ->
+        data
+    end
+  end
+
+  @doc """
+  Retrieves the bids & asks of the order book up to the depth for the given symbol
+
+  ## Example
+  ```
+  {:ok,
+    %Binance.OrderBook{
+      asks: [
+        ["8400.00000000", "2.04078100", []],
+        ["8405.35000000", "0.50354700", []],
+        ["8406.00000000", "0.32769800", []],
+        ["8406.33000000", "0.00239000", []],
+        ["8406.51000000", "0.03241000", []]
+      ],
+      bids: [
+        ["8393.00000000", "0.20453200", []],
+        ["8392.57000000", "0.02639000", []],
+        ["8392.00000000", "1.40893300", []],
+        ["8390.09000000", "0.07047100", []],
+        ["8388.72000000", "0.04577400", []]
+      ],
+      last_update_id: 113634395
+    }
+  }
+  ```
+  """
+  @spec get_depth(String.t(), integer) :: {:ok, %Binance.OrderBook{}} | {:error, error()}
+  def get_depth(symbol, limit) do
+    case HTTPClient.get_binance("/fapi/v1/depth?symbol=#{symbol}&limit=#{limit}") do
+      {:ok, data} -> {:ok, Binance.OrderBook.new(data)}
+      err -> err
+    end
+  end
+
+  # Account
+
+  @doc """
+  Fetches user account balance from binance
+
+  Weight: 1 for a single symbol
+
+  Please read https://binanceapitest.github.io/Binance-Futures-API-doc/trade_and_account/#future-account-balance-user_data
+  """
+  @spec get_balance(map() | nil) :: {:ok, list(%Binance.Futures.Balance{})} | {:error, error()}
+  def get_balance(config \\ nil) do
+    case HTTPClient.get_binance("/fapi/v1/balance", %{}, config) do
+      {:ok, data} ->
+        {:ok, Enum.map(data, &Binance.Futures.Balance.new/1)}
```

Comment:
The API documentation says it returns something like:

```
 {
     "accountId": 26
      "asset": "USDT"
      "balance": "122607.35137903"
      "withdrawAvailable": "102333.54137903"
 }
```

But here we are expecting a list. Is that correct?